web/api/v1/api_test.go

```diff
@@ -1207,6 +1250,56 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, es storage.E
 				},
 			},
 		},
+		{
+			endpoint: api.query,
+			query: url.Values{
+				"query": []string{
+					`label_replace(vector(42), "foo", "bar", "", "") or label_replace(vector(3.1415), "dings", "bums", "", "")`,
+				},
+				"time": []string{"123.4"},
+			},
+			responseAsJSON: `{
+		"resultType": "vector",
+		"result": [
+			{
+				"metric": {
+					"foo": "bar"
+				},
+				"value": [123.4, "42"]
+			},
+			{
+				"metric": {
+					"dings": "bums"
+				},
+				"value": [123.4, "3.1415"]
+			}
+		]
+	}`,
+		},
+		{
+			endpoint: api.queryRange,
+			query: url.Values{
+				"query": []string{"time()"},
+				"start": []string{"0"},
+				"end":   []string{"2"},
+				"step":  []string{"1"},
+				"limit": []string{"1"},
+			},
+			response: &QueryData{
+				ResultType: parser.ValueTypeMatrix,
+				Result: promql.Matrix{
+					promql.Series{
+						Floats: []promql.FPoint{
+							{F: 0, T: timestamp.FromTime(start)},
+							{F: 1, T: timestamp.FromTime(start.Add(1 * time.Second))},
+							{F: 2, T: timestamp.FromTime(start.Add(2 * time.Second))},
+						},
+						// No Metric returned - use zero value for comparison.
```

Comment:
This comment looks out of place. There is a metric returned.

Reply:
I thought the comment referred to the Metric field in promql.Series being empty, not the Floats values. Are Floats considered part of the Metric in this context?

Reply:
OK, I get it now. (What is called "Metric" in this context is essentially the label set.) Still, I wouldn't comment on that. If the returned result doesn't have labels, it's just normal to represent that by the zero value. If a comment explains the obvious, it is confusing because the reader will assume that the comment wants to point to something non-obvious.

Reply:
Makes sense, removed the comment.